### PR TITLE
Make "visual feedback on delete" optional and add it to overlays

### PR DESCRIPTION
**Pull request by crudelios**
There is now an option on the new "Config" window whether to enable the new visual feedback on delete or not.

By default it's disabled, which works like C3 did (apart from a few visual fixes).

Please test to check if everything works, both visually and in terms of behavior.

When active, visual feedback will also show on overlays too.

### CODE REVIEW

File: src/widget/city_overlay_other.c [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #include "city_overlay_other.h"
 
 #include "building/model.h"
+#include "core/config.h"
 #include "city/finance.h"
 #include "core/calc.h"
```
Comment (bvschaik) [inline src/widget/city_overlay_other.c]: Alphabetical order :)
```suggestion
#include "city/finance.h"
#include "core/calc.h"
#include "core/config.h"
```

File: src/widget/city_with_overlay.c [hunk 2]
```diff
@@ -3,6 +3,7 @@
 #include "building/animation.h"
 #include "building/construction.h"
 #include "building/industry.h"
+#include "core/config.h"
 #include "city/view.h"
```
Comment (bvschaik) [inline src/widget/city_with_overlay.c]: Alphabetical order :)
```suggestion
#include "city/view.h"
#include "core/config.h"
```

File: src/widget/city_with_overlay.c [hunk 3]
```diff
@@ -132,26 +133,40 @@ static int is_drawable_farm_corner(int grid_offset)
...
 }
 
-static void draw_flattened_building_footprint(const building *b, int x, int y, int image_offset)
+static int draw_building_as_deleted(building *b)
+{
+    if (!config_get(CONFIG_UI_VISUAL_FEEDBACK_ON_DELETE)) {
+        return 0;
+    }
+    b = building_main(b);
+    return (b->id && (b->is_deleted || map_property_is_deleted(b->grid_offset)));
```
Comment (bvschaik) [inline src/widget/city_with_overlay.c]: Unnecessary parentheses around the whole statement.

File: src/widget/city_with_overlay.c [hunk 4]
```diff
@@ -132,26 +133,40 @@ static int is_drawable_farm_corner(int grid_offset)
...
+}
+
+static int is_multi_tile_terrain(int grid_offset)
+{
+    return (!map_building_at(grid_offset) && map_property_multi_tile_size(grid_offset) > 1);
```
Comment (bvschaik) [inline src/widget/city_with_overlay.c]: Unnecessary parentheses

File: src/widget/city_building_ghost.c [hunk 5]
```diff
@@ -4,6 +4,7 @@
 #include "building/count.h"
 #include "building/industry.h"
 #include "building/properties.h"
+#include "core/config.h"
 #include "city/buildings.h"
 #include "city/finance.h"
 #include "city/view.h"
```
Comment (bvschaik) [inline src/widget/city_building_ghost.c]: Alphabetical order please :)
```suggestion
#include "city/buildings.h"
#include "city/finance.h"
#include "city/view.h"
#include "core/config.h"
```

File: src/widget/city_overlay_other.c [hunk 6]
```diff
@@ -246,6 +247,16 @@ const city_overlay *city_overlay_for_tax_income(void)
...
 }
 
+static int has_deleted_building(int grid_offset)
+{
+    if (!config_get(CONFIG_UI_VISUAL_FEEDBACK_ON_DELETE)) {
+        return 0;
+    }
+    building *b = building_get(map_building_at(grid_offset));
+    b = building_main(b);
+    return (b->id && (b->is_deleted || map_property_is_deleted(b->grid_offset)));
```
Comment (bvschaik) [inline src/widget/city_overlay_other.c]: Unnecessary parentheses around the whole statement.

File: src/widget/city_overlay_other.c (hunk 1 above)
Comment (crudelios) [reply src/widget/city_overlay_other.c]: It seems I don't know the alphabet anymore 😛

**Pull request merged**